action_plugins/_junos_shutdown.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+#
+# Copyright (c) 2017, Juniper Networks Inc. All rights reserved.
+#
+# License: Apache 2.0
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright
+#   notice, this list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright
+#   notice, this list of conditions and the following disclaimer in the
+#   documentation and/or other materials provided with the distribution.
+#
+# * Neither the name of the Juniper Networks nor the
+#   names of its contributors may be used to endorse or promote products
+#   derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY Juniper Networks, Inc. ''AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL Juniper Networks, Inc. BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# Standard library imports
+import os.path
+import sys
+
+# The module_utils path must be added to sys.path in order to import
+# juniper_junos_common. The module_utils path is relative to the path of this
+# file.
+module_utils_path = os.path.normpath(os.path.dirname(__file__) +
+                                     '/../module_utils')
+if module_utils_path is not None:
+    sys.path.insert(0, module_utils_path)
```

Comment:
Why not just use importlib module for this purpose rather adding and deleting path from sys path.

Reply:
@vnitinv 
In the case of the action plugin, you are correct, using importlib would work. I will look at switching this code. In the module itself, importlib won't work, and I guess that's why I forgot it would work in the action plugin.

In the action plugin, the module_utils path really is relative to the action plugin's path. In a module, the code has been packaged and placed in a completely different directory and is running with no context of where the original roles directory is located. I believe that has to be handled by manipulating sys.path. See this:
https://github.com/Juniper/ansible-junos-stdlib/blob/master/library/juniper_junos_facts.py#L166-L210

Reply:
@vnitinv 

Actually, I don't think importlib will work in the case of the action plugin either. The action plugin is not part of a package. So, even though the code to be imported is in a filename that relative to the action plugin (i.e. ../module_utils/juniper_junos_common.py), it can't be imported with importlib because the action plugin is not part of a package and it can't be made part of a package because of Ansible's role packaging requirements.

FWIW, I tried this code:
```
importlib.import_module('..module_utils.juniper_junos_common', '__name__')
```
and it produced this error:
```
ERROR! Unexpected Exception, this is probably a bug: attempted relative import beyond top-level package
```